### ISSUE TITLE
Some Examples do not successfully build on older versions of CUDA whereby cuMemAdvise_v2 and cuMemPrefetchAsync_v2 are not present

### ISSUE THREAD

**Issue by br0kej** [huggingface/candle#2331]
This has been identified on the upstream `cudarc` project in [this issue](https://github.com/coreylowman/cudarc/issues/269) with the associated fix in an open PR [here](https://github.com/coreylowman/cudarc/pull/270). It has not yet been merged yet though.

It's unclear to me what this additional functionality brings and whether it is actually used within Candle. I am happy to open a PR which includes pinning `cuadrc` to `cudarc = { version = "<=0.11.6" }` within  affected `candle-*` crates if folks think it's worthwhile.

**Comment by LaurentMazare** [comment]
Right, I also add the problem on some of my projects and fixed it by explicitely pinning cudarc in my crates but probably better for users not to have to do this so having the constraint in the main `Cargo.toml` file sounds good.

**Comment by br0kej** [comment]
I'll make that change now, run it locally to see if it fixes it and then open the PR 👍🏼

**Comment by br0kej** [comment]
I have opened a PR to resolve this. It was a strange issue as the version was pinned to `0.11.4` but when building Cargo was pulling the latest semver version down anyways. From looking at [this stackoverflow post](https://stackoverflow.com/questions/45224563/how-to-specify-the-exact-version-of-a-dependency) there seems to be some oddness around specifying exact versions!
